integrations/salesforce/oauth.go

```diff
@@ -118,22 +119,34 @@ func (h handler) saveConnection(ctx context.Context, vsid sdktypes.VarScopeID, t
 // so we need to add it on our own. This function also returns the connection's client ID.
 // Based on: https://help.salesforce.com/s/articleView?id=xcloud.remoteaccess_oidc_token_introspection_endpoint.htm
 func (h handler) accessTokenExpiration(ctx context.Context, instanceURL string, t *oauth2.Token, vsid sdktypes.VarScopeID) (string, error) {
-	vs, err := h.vars.Get(ctx, vsid, clientIDVar, clientSecretVar)
+	var clientID, clientSecret string
+
+	vs, err := h.vars.Get(ctx, vsid, common.AuthTypeVar, clientIDVar, clientSecretVar)
 	if err != nil {
 		return "", err
 	}
+	if common.ReadAuthType(vs) == integrations.OAuthPrivate {
```

Comment:
You can do the same here as I suggested in `integration.go`. Either way add an empty line above this line.